asyncpg/introspection.py

```diff
@@ -272,15 +279,15 @@
 
 
 # 'b' for a base type, 'd' for a domain, 'e' for enum.
-SCALAR_TYPE_KINDS = (b'b', b'd', b'e')
+SCALAR_TYPE_KINDS: typing_extensions.Final = (b'b', b'd', b'e')
 
 
-def is_scalar_type(typeinfo) -> bool:
+def is_scalar_type(typeinfo: 'protocol.Record') -> bool:
     return (
         typeinfo['kind'] in SCALAR_TYPE_KINDS and
         not typeinfo['elemtype']
     )
 
 
-def is_domain_type(typeinfo) -> bool:
-    return typeinfo['kind'] == b'd'
+def is_domain_type(typeinfo: 'protocol.Record') -> bool:
+    return typing.cast(bool, typeinfo['kind'] == b'd')
```

Comment:
Is mypy marking this spot as problematic without cast? `==` should be enough to suggest `bool`

Reply:
`mypy` is marking this as returning `Any` from a function that returns `bool`. This happens because `typeinfo['kind']` is `Any`. I think the better option here is to do `typing.cast(bytes, typeinfo['kind']) == b'd'`.